### DISCUSSION TITLE
LDM Notes for May 15th, 2023

### DISCUSSION THREAD

**Discussion by 333fred** [dotnet/csharplang#7205] [LDM Notes]
https://github.com/dotnet/csharplang/blob/main/meetings/2023/LDM-2023-05-15.md

## Agenda

- Breaking Change Warnings
- Primary Constructors

**Comment by DavidArno** [comment]
> There could be an opportunity to create a level between warning and error; diagnostics that appear as errors but are user-suppressible.

This mechanism already exists: analyzers. I can set an analyzer rule to `Error` and it will fail the build, but it's not a "real error" as I can use the warning suppression mechanisms to have it ignored.

**Reply by alrz** [reply]
I'm thinking as long as language versions are compatible with each other (which puts a slight limit on the kinds of breaking changes that could be made), "ahead of time" warnings would be unnecessary? If you don't want to deal with breaks *right now*, you can just stick to `11.0` while taking advantage of the new platform version. I might be wrong but I don't think that would be too limiting in practice, even if the break is not just lexical.
